### PR TITLE
Use a string for the WLM Job ID

### PR DESCRIPTION
**Pull request by roehrich-hpe**
Allow the use of a string for the JobID.

Adjust the printcolumn for JobID so that it is part of the default display for "kubectl get workflows".

### CODE REVIEW

**Comment by NateThornton** [comment]
Could you make that `IntOrString` then it would be backwards compatible?

File: api/v1alpha2/workflow_types.go [hunk 1]
```diff
@@ -105,8 +105,18 @@ type WorkflowSpec struct {
...
 	DesiredState WorkflowState `json:"desiredState"`
 
+	// WLMID identifies the Workflow Manager (WLM), and is set by the WLM
+	// when it creates the workflow resource.
 	WLMID string `json:"wlmID"`
-	JobID int    `json:"jobID"`
+
+	// JobID2 is the WLM job ID that corresponds to this workflow, and is
+	// set by the WLM when it creates the workflow resource.
+	JobID2 string `json:"jobID2"`
```
Comment (bdevcich-hpe) [inline api/v1alpha2/workflow_types.go:114]: We can reuse `JobID`? Can you make the json call it `jobID`?

**Comment by roehrich-hpe** [comment]
> Could you make that `IntOrString` then it would be backwards compatible?

Thanks, @NateThornton , That's working.

**Review by bdevcich-hpe — APPROVED** [review]

**Comment by jameshcorbett** [comment]
To set a string job ID will Flux need to fetch `v1alpha2` or will `v1alpha1` still work?

**Comment by roehrich-hpe** [comment]
> To set a string job ID will Flux need to fetch `v1alpha2` or will `v1alpha1` still work?

You'll need to use v1alpha2.

Keep in mind that it has "alpha" in the name.  v1alpha2 will have more changes before we bump the alpha release number.

**Pull request merged**